cli/popper/utils.py

```diff
@@ -191,9 +182,49 @@ def assert_executable_exists(command):
 
 
 def prettystr(a):
-    if type(a) == DotMap:
+    if isinstance(a, DotMap):
         a = a.toDict()
-    if type(a) == os._Environ:
+    if isinstance(a, os._Environ):
         a = dict(a)
-    if type(a) == dict:
+    if isinstance(a, dict):
         return f'{yaml.dump(a, default_flow_style=False)}'
+
+
+def exec_cmd(cmd, env=None, cwd=os.getcwd(), spawned_processes=set(), stream=True):
```

Comment:
stream usually means that this is a generator function, so maybe we can rename that arg to `log` which indicates that output is to be logged rather than returned